### ISSUE TITLE
declarations of type 'uint' are not supported

### ISSUE THREAD

**Issue by jwoehr** [Qiskit/qiskit-qasm3-import#10]
The openqasm example [adder.qasm](https://github.com/openqasm/openqasm/blob/main/examples/adder.qasm) fails to import with the error:
`qiskit_qasm3_import.exceptions.ConversionError: 24,0: declarations of type 'uint' are not supported
`

**Comment by jakelishman** [comment]
Yeah, it's kind of deliberate at the moment (hence the error message) - Terra doesn't have support for representing any sort of non-const `uint` runtime value, so I had the importer raise errors.

Would it be better if I made the messages say "... because Terra cannot represent them"?

**Comment by jwoehr** [comment]
The example is from an openqasm/openqasm example ... maybe I should file an Issue against the example!
